MacOS compile requires

Compilaltion on MacOS Sonoma 14.2 fails with the following error:
 
```error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: lib/libdeflate.dylib (for architecture arm64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
make: *** [lib/libdeflate.dylib] Error 1
```
Adding the ```-headerpad``` and ```-headerpad_max_install_names``` to LD_LIB_FLAGS fails to resolve the issue, though it does make it much further through compilation, to eventually fail with the error:

```clang: error: unknown argument: '-headerpad'```

--- a/Makefile
+++ b/Makefile
@@ -217,7 +217,7 @@ ifeq ($(shell uname -s),Darwin)
     END_STATIC =
 
     # We need to use special flags to let us rename libraries
-    LD_RENAMEABLE_FLAGS = -headerpad -headerpad_max_install_names
+    LD_RENAMEABLE_FLAGS = -Wl,-headerpad -Wl,-headerpad_max_install_names
 else
     # We are not running on OS X
     $(info OS is Linux)

--- a/Makefile
+++ b/Makefile
@@ -215,6 +215,9 @@ ifeq ($(shell uname -s),Darwin)
     # We don't actually do any static linking on Mac, so we leave this empty.
     START_STATIC =
     END_STATIC =
+
+    # We need to use special flags to let us rename libraries
+    LD_RENAMEABLE_FLAGS = -headerpad -headerpad_max_install_names
 else
     # We are not running on OS X
     $(info OS is Linux)
@@ -252,7 +255,8 @@ else
     # Note that END_STATIC is only safe to use in a mostly-dynamic build, and has to appear or we will try to statically link secret trailing libraries.
     END_STATIC = -Wl,-Bdynamic
 
-
+    # We don't need any flags because we don't need to rename libraries with install_name_tool
+    LD_RENAMEABLE_FLAGS =
 endif
 
 # Set the C++ standard we are using
@@ -656,7 +660,7 @@ ifeq ($(shell uname -s),Darwin)
 endif
 
 $(LIB_DIR)/libdeflate.a: $(LIBDEFLATE_DIR)/*.h $(LIBDEFLATE_DIR)/lib/*.h $(LIBDEFLATE_DIR)/lib/*/*.h $(LIBDEFLATE_DIR)/lib/*.c $(LIBDEFLATE_DIR)/lib/*/*.c
-	+. ./source_me.sh && cd $(LIBDEFLATE_DIR) && V=1 $(MAKE) $(FILTER) && cp libdeflate.a $(CWD)/$(LIB_DIR) && cp libdeflate.h $(CWD)/$(INC_DIR)
+	+. ./source_me.sh && cd $(LIBDEFLATE_DIR) && V=1 LDFLAGS="$(LDFLAGS) $(LD_RENAMEABLE_FLAGS)" $(MAKE) $(FILTER) && cp libdeflate.a $(CWD)/$(LIB_DIR) && cp libdeflate.h $(CWD)/$(INC_DIR)
 
 # We build htslib after libdeflate so it can use libdeflate.
 # We need to do some wizardry to get it to pick up the right build and target system types on modern autotools.

